### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by zarva6596**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,31 @@
...
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let k = '';
+
+    if (this.length === 1) {
+      k += this[0];
+
+      return k;
+    }
+
```
Comment (taniaomelko) [inline src/arrayMethodJoin.js:15]: Це зайве.

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let k = '';
```
Comment (taniaomelko) [inline src/arrayMethodJoin.js:8]: Чого k?

**Review by taniaomelko — CHANGES_REQUESTED** [review]

**Review by taniaomelko — APPROVED** [review]

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let a = 0; a < this.length; a++) {
```
Comment (irenhh) [inline src/arrayMethodJoin.js:10]: usually this variable is called `i`

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,8 +4,25 @@
...
+      if (this[a] === null || this[a] === undefined) {
+        result += separator;
+        continue;
+      }
+
+      result += this[a];
+
+      if (a === this.length - 1) {
```
Comment (irenhh) [inline src/arrayMethodJoin.js:18]: how about adding this to `for` conditions?

**Review by irenhh — CHANGES_REQUESTED** [review]

File: src/arrayMethodJoin.js (hunk 3 above)
Comment (zarva6596) [reply src/arrayMethodJoin.js]: fixed

File: src/arrayMethodJoin.js [hunk 5]
```diff
@@ -7,15 +7,15 @@ function applyCustomJoin() {
...
+      if (this[i] === null || this[i] === undefined) {
         result += separator;
         continue;
       }
 
-      result += this[a];
+      result += this[i];
 
-      if (a === this.length - 1) {
+      if (i === this.length - 1) {
```
Comment (irenhh) [inline src/arrayMethodJoin.js:18]: you didn't fix all the comments

**Review by denys-cheporniuk — CHANGES_REQUESTED** [review]
image
твій метод працює не так як оригінальний метод join

**Review by irenhh — CHANGES_REQUESTED** [review]
check this `[null, 1, 2, 3, undefined]` and compare to the original method

**Review by denys-cheporniuk — APPROVED** [review]

**Pull request closed without merging**